pkg/state/appender.go

```diff
@@ -853,42 +866,62 @@ func (a *txAppender) createCheckerInfo(params *appendBlockParams) (*checkerInfo,
 	return checkerInfo, nil
 }
 
+type initialDiffAndStateHashParams struct {
+	blockHeader               *proto.BlockHeader
+	hasParent                 bool
+	transactions              []proto.Transaction
+	lastSnapshotStateHash     crypto.Digest
+	fixSnapshotsToInitialHash []proto.AtomicSnapshot
+	currentBlockHeight        proto.Height
+	readOnly                  bool
+}
+
+// createInitialDiffAndStateHash creates the initial diff and state hash for the block.
+// It DOES NOT apply the initial snapshot to the state.
+// The diff changes will be cleared after this function.
 func (a *txAppender) createInitialDiffAndStateHash(
-	params *appendBlockParams,
-	hasParent bool,
-	blockInfo *proto.BlockInfo,
+	params initialDiffAndStateHashParams,
 	hasher *txSnapshotHasher,
-) (crypto.Digest, error) {
+) (txSnapshot, crypto.Digest, error) {
 	// Create miner balance diff.
 	// This adds 60% of prev block fees as very first balance diff of the current block in case NG is activated.
 	// Before NG activation it adds all transactions fees to the miner's balance.
-	minerAndRewardDiff, err := a.blockDiffer.createMinerAndRewardDiff(params.block, hasParent, params.transactions)
+	minerAndRewardDiff, err := a.blockDiffer.createMinerAndRewardDiff(
+		params.blockHeader,
+		params.hasParent,
+		params.transactions,
+	)
 	if err != nil {
-		return crypto.Digest{}, err
+		return txSnapshot{}, crypto.Digest{}, err
 	}
 
 	// create the initial snapshot
 	initialSnapshot, err := a.txHandler.tp.createInitialBlockSnapshot(minerAndRewardDiff.balancesChanges())
 	if err != nil {
-		return crypto.Digest{}, errors.Wrap(err, "failed to create initial snapshot")
+		return txSnapshot{}, crypto.Digest{}, errors.Wrap(err, "failed to create initial snapshot")
 	}
 
-	currentBlockHeight := blockInfo.Height
-
+	// TODO: is it necesary to perform here any actions with diff storage??
```

Comment:
Still TODO. What do you thing, @Anton-Rampage ?
Should it be removed or fixed (TODO looks like a question)?